pipelines/scripts/test/test_deploy_odw_package.py

```diff
@@ -0,0 +1,327 @@
+            }
+        }
+    ]
+    expected_odw_packages = [
+        {
+            "name": "odw-1.0.0-some_commit-py3-none-any.whl",
+            "properties": {
+                "uploadedTimestamp": "2025-08-01T13:58:54.3370407+00:00"
+            }
+        },
+        {
+            "name": "odw-1.0.0-some_other_other_commit-py3-none-any.whl",
+            "properties": {
+                "uploadedTimestamp": "2025-08-02T13:58:54.3370407+00:00"
+            }
+        },
+        {
+            "name": "odw-1.0.0-some_other_commit-py3-none-any.whl",
+            "properties": {
+                "uploadedTimestamp": "2025-08-05T13:58:54.3370407+00:00"
+            }
+        }
+    ]
+    with mock.patch("pipelines.scripts.util.synapse_workspace_manager.SynapseWorkspaceManager") as mock_workspace_manager:
+        mock_workspace_manager.get_workspace_packages.return_value = mock_get_workspace_packages
+        actual_odw_packages = ODWPackageDeployer().get_existing_odw_wheels(mock_workspace_manager)
+        assert actual_odw_packages == expected_odw_packages
+
+
+def test_upload_new_wheel__with_no_existing_package():
+    """
+        Test uploading an odw package for the first time
+    """
+    env = "mock_env"
+    wheel_name = "odw_test_wheel.whl"
+    def get_spark_pool(inst, pool_name: str):
+        if pool_name == "pinssynspodwpr":
+            return {
+                "properties": {
+                    "customLibraries": [
+                        {
+                            "name": "some_wheel.whl",
+                            "path": f"pins-synw-odw-{env}-uks/libraries/some_wheel.whl",
+                            "containerName": "prep",
+                            "uploadedTimestamp": "0001-01-01T00:00:00+00:00",
+                            "type": "whl"
+                        },
+                        {
+                            "name": "some_other_wheel.whl",
+                            "path": f"pins-synw-odw-{env}-uks/libraries/some_other_wheel.whl",
+                            "containerName": "prep",
+                            "uploadedTimestamp": "0001-01-01T00:00:00+00:00",
+                            "type": "whl"
+                        },
+                        {
+                            "name": "odw-old-wheel.whl",
+                            "path": f"pins-synw-odw-{env}-uks/libraries/odw-old-wheel.whl",
+                            "containerName": "prep",
+                            "uploadedTimestamp": "0005-01-01T00:00:00+00:00",
+                            "type": "whl"
+                        }
+                    ]
+                }
+            }
+        if pool_name == "pinssynspodw34":
+            return {
+                "properties": dict()
+            }
+        pytest.fail(f"Unexpected pool name '{pool_name}' used in the test - please review this")
+    mock_odw_wheels = []  # No other odw wheels
```

Comment:
Have you thought using parametrize (pytest.mark.parametrize) for the variations of this one ? could reduce the test boilerplate as they repeat a similar structure with different inputs